neuralmonkey/dataset.py

```diff
@@ -1,106 +1,251 @@
 
-        series: Dictionary from the series name to the actual data.
+    dataset = Dataset(name, series, series_outputs)
+    log("Dataset length: {}".format(len(dataset)))
 
-        series_outputs: Output files for target series.
+    return dataset
 
-        random_seed: Random seed used for shuffling.
 
+def _get_name_from_paths(series_paths: Dict[str, str]) -> str:
+    """Construct name for a dataset using the paths to its files.
+
+    Arguments:
+        series_paths: A dictionary which maps serie names to the paths
+                      of their input files.
+
+    Returns:
+        The name for the dataset.
     """
+    name = "dataset"
+    for _, path in series_paths.items():
+        name += "-{}".format(path)
+    return name
 
-    def __init__(self, name, series, series_outputs, random_seed=None):
-        """
-        Creates a dataset from the provided already preprocessed series of
-        data.
 
-        Args:
+def _get_series_paths(kwargs: Dict[str, str]) -> Dict[str, str]:
+    """Get paths to files that contain data from the dataset keyword
+    argument specs.
+
+    Input file for a serie named 'xxx' is specified by parameter 's_xxx'
+
+    Arguments:
+        kwargs: A dictionary containing the dataset keyword argument specs.
+
+    Returns:
+        A dictionary which maps serie names to the paths of their input files.
+    """
+    keys = [k for k in list(kwargs.keys()) if SERIES_SOURCE.match(k)]
+    names = [SERIES_SOURCE.match(k).group(1) for k in keys]
+
+    return {name : kwargs[key] for name, key in zip(names, keys)}
+
+
+def _get_series_outputs(kwargs: Dict[str, str]) -> Dict[str, str]:
+    """Get paths to series outputs from the dataset keyword argument specs.
+    Output file for a series named 'xxx' is specified by parameter 's_xxx_out'
+
+    Arguments:
+        kwargs: A dictionary containing the dataset keyword argument specs.
+
+    Returns:
+        A dictionary which maps serie names to the paths for their output files.
+    """
+    return {SERIES_OUTPUT.match(key).group(1): value
+            for key, value in kwargs.items() if SERIES_OUTPUT.match(key)}
 
-        """
 
+def create_dataset_series(path: str,
+                          preprocess: Callable[[str], str]) -> Iterable:
+    """Create dataset series.
+
+    Arguments:
+        path: The path of the file with the data
+        preprocess: Preprocessor function
+
+    Returns:
+        The dataset series.
+    """
+    log("Loading {}".format(path))
+    file_type = magic.from_file(path, mime=True)
+
+    if file_type.startswith('text/'):
+        reader = PlainTextFileReader(path)
+        for line in reader.read():
+            yield preprocess(line)
+    elif file_type == 'application/octet-stream':
+        return np.load(path)
+    else:
+        raise Exception("Unsupported data type: {}, file {}"
+                        .format(file_type, path))
+
+
+
+class Dataset(collections.Sized):
```

Comment:
:-) To chtěl ten mypy aby to bylo sized.. Cokoli co má __len__ má dědit od Sized.. Má to takhle už i vocabulary..